registry/repositories/user.interface.ts

```diff
@@ -0,0 +1,25 @@
+export interface UserRepository {
+  getUser: (username: string) => Promise<User | null>;
+  getUserById: (id: string) => Promise<User | null>;
+  createAuthBasicUser: (props: AuthBasicSignup) => Promise<User>;
+}
+
+export type User = {
+  userId: string;
+  username: string;
+  password: string;
+  createdAt: string;
+};
+
+declare global {
+  namespace Express {
+    interface Request {
+      user?: User;
```

Comment:
Noice! Do this require any ts config changes?

Reply:
Nope, it works without changes, you can access the object with `req.user`